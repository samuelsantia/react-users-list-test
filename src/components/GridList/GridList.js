import styled from 'styled-components/macro';

const GridList = styled.section`
  display: grid;
  margin: var(--grid-margin, 1rem);
  grid-template-columns: var(--grid-columns, 1fr);
  gap: var(--grid-gap, 1rem);
`;

export default GridList;
