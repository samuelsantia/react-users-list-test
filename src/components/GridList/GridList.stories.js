import React from 'react';

import GridList from './GridList';
import Card from '../Card';
import CardHeader, { bgColors } from '../CardHeader/CardHeader';
import CardContent from '../CardContent';
import Typography from '../Typography';
import ProfileAvatar from '../ProfileAvatar';

import users from './users.fixture';

const colors = Object.keys(bgColors);

const ProfileCard = ({ text, avatar, userName, bgc }) => (
  <Card>
    <CardHeader bgc={bgc}>
      <ProfileAvatar src={avatar} alt={userName} />
    </CardHeader>
    <CardContent>
      <Typography variant="h5" tag="h2" weight="normal">
        {userName}
      </Typography>
      <Typography variant="body2">{text}</Typography>
    </CardContent>
  </Card>
);

export default {
  title: 'GridList',
  component: GridList,
};

export const basic = () => (
  <GridList>
    {users.map(({ id, first_name, last_name, email, avatar }, idx) => (
      <ProfileCard
        key={id}
        userName={`${first_name} ${last_name}`}
        text={email}
        avatar={avatar}
        bgc={colors[idx]}
      />
    ))}
  </GridList>
);
