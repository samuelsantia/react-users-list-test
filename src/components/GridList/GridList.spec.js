import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import GridList from './GridList';

describe('<GridList />', () => {
  it('should render', () => {
    const { container } = render(<GridList>Test GridList</GridList>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
