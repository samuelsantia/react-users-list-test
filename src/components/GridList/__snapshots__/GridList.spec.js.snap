// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridList /> should render 1`] = `
.c0 {
  display: grid;
  margin: var(--grid-margin,1rem);
  grid-template-columns: var(--grid-columns,1fr);
  gap: var(--grid-gap,1rem);
}

<section
  class="c0"
>
  Test GridList
</section>
`;
